includes/specials/SpecialUserrights.php

```diff
@@ -21,18 +21,79 @@
  * @ingroup SpecialPage
  */
 
+use Wikia\Service\User\Permissions\PermissionsServiceAccessor;
+
 /**
  * Special page to allow managing user group membership
  *
  * @ingroup SpecialPage
  */
 class UserrightsPage extends SpecialPage {
+	use PermissionsServiceAccessor;
 	# The target of the local right-adjuster's interest.  Can be gotten from
 	# either a GET parameter or a subpage-style parameter, so have a member
 	# variable for it.
 	protected $mTarget;
 	protected $isself = false;
 
+	public static function userCanChangeRights( \User $user, $isself, $checkIfSelf = true ) {
+		$available = $user->changeableGroups();
+		return !empty( $available['add'] )
+		|| !empty( $available['remove'] )
+		|| ( ( $isself || !$checkIfSelf ) &&
+			( !empty( $available['add-self'] )
+				|| !empty( $available['remove-self'] ) ) );
+	}
+
+	/**
+	 * Add a rights log entry for an action.
+	 * @param User $user
+	 * @param $oldGroups
+	 * @param $newGroups
+	 * @param $reason
+	 */
+	public static function addLogEntry( \User $user, $oldGroups, $newGroups, $reason ) {
+		$log = new LogPage( 'rights' );
+
+		$log->addEntry( 'rights',
+			$user->getUserPage(),
+			$reason,
+			array(
+				self::makeGroupNameListForLog( $oldGroups ),
+				self::makeGroupNameListForLog( $newGroups )
+			)
+		);
+	}
+
+	public static function isGlobalNonEditableGroup( $globalGroups, $group ) {
```

Comment:
The `$globalGroups` argument is a static list so we could fetch it ourselves instead of requesting callers to do that in every single place where this method is called